src/gluonts/mx/block/regularization.py

```diff
@@ -0,0 +1,179 @@
+    year={2017}
+    }
+    Parameters
+    ----------
+    alpha : float, default 0
+        The scaling coefficient of the regularization.
+    weight : float or None
+        Global scalar weight for loss.
+    batch_axis : int, default 0
+        The axis that represents mini-batch.
+    time_axis : int, default 0
+        The axis that represents time-step.
+    """
+
+    @validated()
+    def __init__(
+        self,
+        alpha: float = 0,
+        weight: Optional[float] = None,
+        batch_axis: int = 1,
+        time_axis: int = 0,
+        **kwargs
+    ):
+        super(ActivationRegularizationLoss, self).__init__(
+            weight, batch_axis, **kwargs
+        )
+        self._alpha = alpha
+        self._batch_axis = batch_axis
+        self._time_axis = time_axis
+
+    def __repr__(self):
+        s = "ActivationRegularizationLoss (alpha={alpha})"
+        return s.format(alpha=self._alpha)
+
+    def hybrid_forward(self, F, *states: List[Tensor]) -> Tensor:
+        """
+        Parameters
+        ----------
+        states : list
+            the stack outputs from RNN, which consists of output from each time step (TNC).
+        Returns
+        --------
+        loss : NDArray
+            loss tensor with shape (batch_size,). Dimensions other than batch_axis are averaged out.
+        """
+        if self._alpha != 0:
+            if states:
+                means = []
+                for state in states:
+                    if isinstance(state, list):
+                        state = F.stack(*state, axis=self._time_axis)
+                    means.append(
+                        self._alpha
+                        * state.__pow__(2).mean(
+                            axis=self._batch_axis, exclude=True
+                        )
+                    )
+                return F.add_n(*means)
+            else:
+                return F.zeros(1)
+        return F.zeros(1)
+
+
+class TemporalActivationRegularizationLoss(Loss):
+    r"""Computes Temporal Activation Regularization Loss. (alias: TAR)
+    The formulation is as below:
+    .. math::
+    L = \beta L_2(h_t-h_{t+1})
+    where :math:`L_2(\cdot) = {||\cdot||}_2, h_t` is the output of the RNN at timestep t,
+    :math:`h_{t+1}` is the output of the RNN at timestep t+1, :math:`\beta` is scaling coefficient.
+    The implementation follows the work::
+    @article{merity2017revisiting,
+    title={Revisiting Activation Regularization for Language RNNs},
+    author={Merity, Stephen and McCann, Bryan and Socher, Richard},
+    journal={arXiv preprint arXiv:1708.01009},
+    year={2017}
+    }
+    Parameters
+    ----------
+    beta : float, default 0
+        The scaling coefficient of the regularization.
+    weight : float or None
+        Global scalar weight for loss.
+    batch_axis : int, default 0
+        The axis that represents mini-batch.
+    time_axis : int, default 0
+        The axis that represents time-step.
```

Comment:
Same as above here: additional space between variable names and `:`; remove explicit type for input parameters.